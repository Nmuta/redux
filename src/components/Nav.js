import { Component } from "react";
import {updateMessage} from "../actions/updateMessage";
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
    


class Nav extends Component{

    state = {localMessage: "not set yet"}

    updateLocalWord = (e) => {
        this.setState({localMessage: e.target.value})
        console.log(this.state.localMessage)
    }

    sendToRedux = (e) => {
        e.preventDefault();
        this.props.updateMessage(this.state.localMessage)
    }

    doStuff = (e) => {
        e ? e.preventDefault(): null;
        return "things are here";
    }

    
    render(){
        return(
            <div className='nav'>
                  <h5 className="learn">Learn ReAcT</h5>
                  <button className="btn" onClick={this.doStuff}>do stuff </button>
                  <span className="countable"> I am countable </span>
                  <span className="countable"> I am countable </span>
                  <span className="countable"> I am countable </span>
                 <input onChange={this.updateLocalWord}/>
                 <button onClick={this.sendToRedux}>submit</button>
            </div>
            )
    }
}


const mapDispatchToProps = dispatch => bindActionCreators({updateMessage}, dispatch)

export default connect(
    null,
    mapDispatchToProps
  )(Nav);