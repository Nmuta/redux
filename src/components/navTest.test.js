import Nav from './Nav';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

const middleton = [thunk];
const mockStore = configureMockStore(middleton);


test('Shallow test of the Nav component', ()=> {
    const wrapper = shallow(
          <Nav.WrappedComponent />
        )

    const countable_objects = wrapper.find('.countable');
    expect(countable_objects).toHaveLength(3);

    const clickable_button = wrapper.find('.btn');
    console.log("clickable button is ", clickable_button.get(0))
    

})

test('renders the same way every time', () => {
    const tree = renderer.create( <Nav.WrappedComponent /> ).toJSON();
    // expect(tree).toMatchSnapshot();
  });

