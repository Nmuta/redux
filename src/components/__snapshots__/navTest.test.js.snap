// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the same way every time 1`] = `
<div
  className="nav"
>
  <h5
    className="learn"
  >
    LeArn ReAcT
  </h5>
  <button
    className="btn"
    onClick={[Function]}
  >
    do stuff 
  </button>
  <span
    className="countable"
  >
     I am countable 
  </span>
  <span
    className="countable"
  >
     I am countable 
  </span>
  <span
    className="countable"
  >
     I am countable 
  </span>
  <input
    onChange={[Function]}
  />
  <button
    onClick={[Function]}
  >
    submit
  </button>
</div>
`;
